patches/server/0164-Add-PlayerArmorChangeEvent.patch

```diff
@@ -20,7 +20,7 @@ index 8044082ed3ca6076af38e4299e50f1f690d02a72..0dc03f53c80aac91a914bea91958a6c1
  
              if (!ItemStack.matches(itemstack1, itemstack)) {
 +                // Paper start - PlayerArmorChangeEvent
-+                if (this instanceof ServerPlayer && enumitemslot.getType() == EquipmentSlot.Type.ARMOR) {
++                if (this instanceof ServerPlayer && enumitemslot.getType() == EquipmentSlot.Type.ARMOR && !((ServerPlayer) this).joining) {
```

Comment:
This probably isn't the right solution. It looks like there are other instances where collectEquipmentChanges could be called. Once place I see, is spectating an entity that isn't tracking the player, I couldn't think of when that might happen, but the way the code is written, that would *also* fire this event.

Reply:
Honestly forgot I made this PR haha, but yeah;  after initially figuring out how the event works it is inherently flawed.  I only planned for this PR to fix when this event occurs on player join, but it might be a good idea to have some discussion for another PR that defines exactly when this event should fire.

Reply:
So I can think of two possible solutions, well one fully defined one, one more complicated.
1. Add a boolean parameter to the 2 methods in LivingEntity, `detectEquipmentChanges`, and `collectEquipmentChanges` to determine if the event should be called.
2. Find a better way fix to the issue spigot fixed which caused this event to fire in the wrong spot.